app/scripts/modules/core/cache/cacheInitializer.service.spec.ts

```diff
@@ -0,0 +1,156 @@
+                          _infrastructureCaches_: InfrastructureCacheService,
+                          _accountService_: AccountService,
+                          _securityGroupReader_: any,
+                          _applicationReader_: ApplicationReader,
+                          _igorService_: any) {
+      $q = _$q_;
+      $root = _$rootScope_;
+      cacheInitializer = _cacheInitializer_;
+      infrastructureCache = _infrastructureCaches_;
+      accountService = _accountService_;
+      securityGroupReader = _securityGroupReader_;
+      applicationReader = _applicationReader_;
+      igorService = _igorService_;
+    })
+  );
+
+  it('should initialize injected dependencies', function () {
+    expect($q).toBeDefined();
+    expect($root).toBeDefined();
+    expect(cacheInitializer).toBeDefined();
+    expect(accountService).toBeDefined();
+    expect(securityGroupReader).toBeDefined();
+    expect(applicationReader).toBeDefined();
+    expect(igorService).toBeDefined();
+  });
+
+  describe('suite title', () => {
+
+    const keys: IKeys = {
+      account: ['account1', 'account2', 'account3'],
+      sg: ['sg1', 'sg2', 'sg3'],
+      app: ['app1', 'app2', 'app3'],
+      bm: ['bm1', 'bm2', 'bm3']
+    };
+
+    beforeEach(() => {
+      spyOn(accountService, 'listAccounts').and.returnValue($q.when(keys.account));
+      spyOn(securityGroupReader, 'getAllSecurityGroups').and.returnValue($q.when(keys.sg));
+      spyOn(applicationReader, 'listApplications').and.returnValue($q.when(keys.app));
+      spyOn(igorService, 'listMasters').and.returnValue($q.when(keys.bm));
+    });
+
+    it('should initialize the cache initializer with the initialization values', () => {
+      cacheInitializer.initialize().then((result: any[]) => {
+
+        expect(result.length).toBe(13); // from infrastructure cache config
+        const flattened: string[][] = flatten(result); // only the arrays that actually contain data
+        expect (flattened.length).toBe(4); // the four initialized string[] above used for the spyOns.
+
+        const prefixes: string[] = ['account', 'sg', 'app', 'bm'];
+        const foundKeys: IFoundKeys = {
+          account: false,
+          sg: false,
+          app: false,
+          bm: false
+        };
+
+        // verify the contents of each array; that they match the values used for initialization
+        for (let i = 0; i < 4; i++) {
+
+          const item: string[] = flattened[i];
+          for (let j = 0; j < 4; j++) {
+
+            const prefix: string = prefixes[j];
+            if (item[0].startsWith(prefix)) {
+              expect(item).toEqual(keys[prefix]);
+              foundKeys[prefix] = true;
+            }
+          }
+        }
+
+        // finally check the boolean object to confirm each key was found
+        Object.keys(foundKeys).forEach((key: string) => expect(foundKeys[key]).toBeTruthy());
+      });
+      $root.$digest();
+    });
```

Comment:
I've burned myself with tests like this before, where all the `expect` calls are in a promise. If the promise doesn't resolve, the test itself will still pass.

It's probably worth adding a check that things got tested, e.g., at the beginning of the spec (before the promise call starts):
```javascript
let initialized = false;
```
then, after the `Object.keys(foundKeys)` call:
```javascript
initialized = true;
```
and, finally, after `$root.$digest();`:
```javascript
expect(initialized).toBe(true);
```
That way, if something goes awry in the `cacheInitializer.initialize()` call, the test will still fail.